python/taichi/lang/ops.py

```diff
@@ -399,108 +627,218 @@ def random(dtype=float):
+
+    Do not confuse this with the matrix multiplication `x1 @ x2` if
+    both `x1` and `x2` are matrices.
 
     Args:
-        a (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): A number or a matrix.
-        b (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): A number or a matrix.
+        x1, x2 (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): \
+            The arguments to be multiplied. If both `x1`, `x2` are matrices they must have the \
+                same shape.
 
     Returns:
-        `a` multiplied by `b`.
+        The product of `x1` and `x2`, element-wise. This is a scalar if both `x1` and `x2` are scalars.
+
+    Example::
+
+        >>> @ti.kernel
+        >>> def main():
+        >>>     x = ti.Matrix([0, 1, 2])
+        >>>     y = ti.Matrix([-1, 0, 1])
+        >>>     z = ti.mul(x, y)
+        >>>     print(z)
+        >>>
+        >>> main()
+        [0, 0, 2]
     """
-    return _binary_operation(_ti_core.expr_mul, _bt_ops_mod.mul, a, b)
+    return _binary_operation(_ti_core.expr_mul, _bt_ops_mod.mul, x1, x2)
 
 
 @binary
-def mod(a, b):
-    """The remainder function.
+def mod(x1, x2):
+    """Returns the element-wise remainder of division.
+
+    This is equivalent to the Python modulus operator `x1 % x2` and
+    has the same sign as the divisor x2.
 
     Args:
-        a (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): A number or a matrix.
-        b (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): A number or a matrix with elements not equal to zero.
+        x1 (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): \
+            Dividend scalar or matrix.
+
+        x2 (Union[:class:`~taichi.lang.expr.Expr`, :class:`~taichi.lang.matrix.Matrix`]): \
+            Divisor scalar or matrix. If both `x1` and `x2` are matrices they must have the same shape.
 
     Returns:
-        The remainder of `a` divided by `b`.
+        The element-wise remainder of the quotient `floordiv(x1, x2)`. This is a scalar \
+            if both `x1` and `x2` are scalars.
+
+    Example::
+
+        >>> @ti.kernel
+        >>> def main():
+        >>>     x = ti.Matrix([3.0, 4.0, 5.0])
+        >>>     y = 3
+        >>>     z = ti.mod(y, x)
+        >>>     print(z)
+        >>>
+        >>> main()
+        [1.0, 0.0, 4.0]
     """
     def expr_python_mod(a, b):
         # a % b = a - (a // b) * b
         quotient = expr.Expr(_ti_core.expr_floordiv(a, b))
         multiply = expr.Expr(_ti_core.expr_mul(b, quotient.ptr))
         return _ti_core.expr_sub(a, multiply.ptr)
 
-    return _binary_operation(expr_python_mod, _bt_ops_mod.mod, a, b)
+    return _binary_operation(expr_python_mod, _bt_ops_mod.mod, x1, x2)
 
 
 @binary
-def pow(a, b):  # pylint: disable=W0622
-    """The power function.
+def pow(x1, x2):  # pylint: disable=W0622
```

Comment:
I personally find that math formulas are more powerful than words here. :P Especially for these ops

Reply:
Math formulas may not be exact here. `x^a` could be interpreted as `x xor a`.

Reply:
I meant "real" math formulas :P https://pytorch.org/docs/stable/generated/torch.pow.html?highlight=pow#torch.pow sphinx should support this in some way